health-checker/src/main/java/io/stargate/health/DataStoreHealthChecker.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.health;
+
+import com.codahale.metrics.health.HealthCheck;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.datastore.ResultSet;
+import io.stargate.db.datastore.Row;
+import java.util.UUID;
+import java.util.concurrent.Future;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DataStoreHealthChecker extends HealthCheck {
+  private static final Logger logger = LoggerFactory.getLogger(DataStoreHealthChecker.class);
+
+  private final BundleContext context;
+
+  public DataStoreHealthChecker(BundleContext context) {
+    this.context = context;
+  }
+
+  @Override
+  protected Result check() {
+    try {
+      ServiceReference<DataStoreFactory> dataStoreFactoryReference =
+          context.getServiceReference(DataStoreFactory.class);
+      if (dataStoreFactoryReference != null) {
+        DataStoreFactory dataStoreFactory = context.getService(dataStoreFactoryReference);
+        try {
+          DataStore dataStore = dataStoreFactory.createInternal();
+
+          Future<ResultSet> rs =
```

Comment:
What consistency level this query uses execution? Is it `QUORUM`? If it is `ONE` it may be too low for health-check.

Reply:
Good point. `system.local` is handled by Stargate itself, so running this query does not guarantee that we can fetch data from storage nodes. On the other hand changing this check to hit actual storage nodes may be too much for this PR, which only deals with exposing more of exisisting checks to HTTP clients. I'll file a follow-up issues for this.

Reply:
filed #631